### PR TITLE
Add minimal set of tests for GLAM Fenix queries

### LINKED ISSUE #1459: Write minimal tests for Glean GLAM ETL scripts
There are no tests for glean GLAM ETL scripts, which makes it difficult to determine whether the scripts are working as intended. A set of tests would be useful.

### PR DESCRIPTION
**Pull request by acmiyaguchi**
This fixes #1459.

### CODE REVIEW

File: script/entrypoint
Comment (acmiyaguchi) [inline script/entrypoint:22]: This was causing test failures because of mypy or pytest-mypy, which is due to the free floating `data.py` files in each of the sql tests being treated as a duplicate module (https://app.circleci.com/pipelines/github/mozilla/bigquery-etl/6996/workflows/ad741abd-36a0-4649-bf39-1e8e0d1616b1/jobs/35787).

File: tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_bucket_counts_v1/test_minimal/data.py
Comment (acmiyaguchi) [inline tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_bucket_counts_v1/test_minimal/data.py:50]: The behavior of range_min and range_max is not intuitive, there might be something weird going on here.

File: tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_percentiles_v1/test_minimal/data.py
Comment (acmiyaguchi) [inline tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_percentiles_v1/test_minimal/data.py:50]: This is the most egregious error, there should be at most 2 users.

**Review by acmiyaguchi — COMMENTED** [review]
There are a couple of issues that should be sorted out.

File: tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_percentiles_v1/test_minimal/data.py
Comment (Ben-Wu) [reply tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_percentiles_v1/test_minimal/data.py:50]: Looking at the query it looks like this is because of the cross join with the `*` combos.  In the final query you should be able to do `COUNT(DISTINCT(client_id)) AS total_users`

File: tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_probe_counts_v1/test_minimal/data.py
Comment (Ben-Wu) [inline tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_probe_counts_v1/test_minimal/data.py:31]: Could you use a fake bucket count and range to make this simpler?

File: tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__clients_histogram_aggregates_v1/test_minimal/data.py
Comment (Ben-Wu) [inline tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__clients_histogram_aggregates_v1/test_minimal/data.py:25]: Although not exactly minimal, it would be more interesting if the test summed onto an existing aggregate

**Review by Ben-Wu — COMMENTED** [review]
This is useful and it helped me understand the etl better.

I don't really understand what's happening with scalar bucket counts though.

File: script/entrypoint
Comment (acmiyaguchi) [reply script/entrypoint:22]: I think it has to do with pytest collecting all of the python files under the test directory as individual modules (although this is mostly conjecture). Since I have a `data.py` file in each test, these appear to be the same module several times. When run outside of pytest, there is no `__init__.py` file and therefore it's ignored. 

If I add a `__init__.py` file to each test going up the directory levels so each test appears like a unique module, then the pytest mypy plugin will work correctly. I want these scripts to be standalone, not part of a module though.

File: tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__histogram_percentiles_v1/test_minimal/data.py
Comment (acmiyaguchi) [reply tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__histogram_percentiles_v1/test_minimal/data.py:15]: This is a fair assessment; I found it easier to copy and paste while putting these together, but it would be easier to maintain with a smaller set.

File: tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_percentiles_v1/test_minimal/data.py
Comment (acmiyaguchi) [reply tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_percentiles_v1/test_minimal/data.py:50]: Thanks for catching this, this fixes the user counts.

File: tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__clients_histogram_aggregates_v1/test_minimal/data.py
Comment (acmiyaguchi) [reply tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__clients_histogram_aggregates_v1/test_minimal/data.py:25]: Added a separate test summing onto an existing aggregate.

File: tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_bucket_counts_v1/test_minimal/data.py
Comment (Ben-Wu) [reply tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_bucket_counts_v1/test_minimal/data.py:50]: Can clients_scalar_aggregates have `*` values for ping_type and os?  Could this be caused by the cross join with the static combos?  Maybe not because app_build_id is defined.  If I understand correctly, min/max should be 2/3 right?

**Review by Ben-Wu — COMMENTED** [review]
I think the range in scalar bucket counts should be figured out before merging.  But otherwise this is looks good to me.

File: tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_bucket_counts_v1/test_minimal/data.py
Comment (acmiyaguchi) [reply tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_bucket_counts_v1/test_minimal/data.py:50]: Even with the cross join, I'm not sure how the range max (in log space) could be set to 5 when the max value is 8.

Comment (Ben-Wu) [reply tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_bucket_counts_v1/test_minimal/data.py:50]: I have a feeling it might be related to the cross join and the input data having `"ping_type": "*", "os": "*",` but I don't really know.  If you can't figure out then I don't want to block merging this.  It would be nice to verify in case there's actually an issue in the etl though.

File: script/entrypoint
Comment (acmiyaguchi) [reply script/entrypoint:22]: This seems to work, I'll opt to use the glob ignore option.

**Review by Ben-Wu — APPROVED** [review]
I think this is acceptable with the range issue if we can't figure it out at all but maybe create an issue so we don't forget (although that might still happen).

File: tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_bucket_counts_v1/test_minimal/data.py
Comment (acmiyaguchi) [reply tests/sql/glam-fenix-dev/glam_etl/org_mozilla_fenix_glam_nightly__scalar_bucket_counts_v1/test_minimal/data.py:50]: It's tangentially related to the cross join. I'm going to follow up with this in a separate PR because this one is already very large for a minimal PR.

**Pull request merged**
